server.js

```diff
@@ -15,13 +37,77 @@ const app = express();
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
+const listEndpoints = require("express-list-endpoints")
+
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(listEndpoints(app));
 });
 
+// GET - fetch latest 20 thoughts 
+app.get('/thoughts', async (req, res) => {
+  try{
+    const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec(); //execute
+    res.status(200).json({
+      success: true,
+      response: thoughts,
+      message: "Thoughts retreived"
+  })
+} catch (err) {
+    res.status(400).json({
+      success: false,
+      response: err,
+      message: "Error, no thoughts found"
+  })
+}
+})
+
+// POST - post a new thought 
+app.post('/thoughts', async (req, res) => {
+  const { message } = req.body; // removed createdAt so that the user cannot input the date, and it will always set to default
+  const thoughts = new Thought({ message, hearts: 0 }); // hearts always 0, but can still update using patch request 
+  try {
+    const savedThought = await thoughts.save();
+    res.status(201).json(savedThought);
+  }
+  catch(err) {
+    res.status(400).json({
+      message: "Could not save thought to the database", error: err.errors
+    })
+  };
+})
```

Comment:
In your GET endpoint, you have a more comprehensive response for both the successful attempt and the failed attempt (success, response, and message). I think Daniel has said that it's good practice to have the same response structure for all endpoints. So, the POST (and PATCH) should have the same response structures.